src/tests/utils.cairo

```diff
@@ -12,20 +12,11 @@ fn deploy(contract_class_hash: felt252, calldata: Array<felt252>) -> ContractAdd
     address
 }
 
-/// Pop the earliest unpopped logged event for the contract as the requested type
-/// and checks there's no more keys or data left on the event, preventing unaccounted params.
-///
-/// This function also removes the first key from the event, to match the event
-/// structure key params without the event ID.
-///
-/// This method doesn't currently work for components events that are not flattened
-/// because an extra key is added, pushing the event ID key to the second position.
+/// Pop the earliest unpopped logged event enum for the contract and checks
+/// there's no more keys or data left on the event, preventing unaccounted params.
```

Comment:
```suggestion
/// there's no more keys or data left on the event, preventing unaccounted params.
///
/// NOTE: If the event enum contains two `flat` events with the same structure (member types),
/// this function will always match the first event, even when the second one is emitted.
```

Reply:
This is related to the implementation of the event enum deserialization for flat events. You can find code for checking this [behavior here](https://github.com/starkware-libs/cairo/blob/eb75434a5e8c7ea1f4f97591b97217859dc7e621/crates/cairo-lang-starknet/src/plugin/plugin_test_data/contracts/events#L328).

Reply:
Awesome! Thank you, sir